### ISSUE TITLE
Exception encountered upon retrieval of relmongo annotated entity from repository

### ISSUE THREAD

**Issue by mancini0** [kaiso/relmongo#11]
When creating some "app" entities,  and assigning a few to a user entity, 

<app1 = new Application(...), app2=new Application(...), app3 =... omitted for brevity>

```
    User user1 = User.builder()
        .uuid("uuid1")
        .name("Don Johnson")
        .app(app1)
        .app(app3)
        .build();

    appRepo.save(app1);
    appRepo.save(app2);
    appRepo.save(app3);
    userRepo.save(user1);
```

My user entity:
```
@Data
@Document
@Builder
public class User {
  @Id 
  private String uuid;
  private String name;
  @OneToMany(fetch=FetchType.EAGER)
  @JoinProperty(name="apps")
  @Singular
  private Set<Application> apps;
}
```

My app entity:

```
@Data
@Document
@Builder
public class Application {
  @Id
  //@NotBlank
  //@Length(max =250)
  private String uuid;
  //@NotBlank
  @Length(max=50)
  private String name;
  private String version;
  private Date updatedAt;
```

Without the @EnableRelationalMongo annotation, I see the apps are persisted as expected, using spring data rest I see my 3 apps at the /apps endpoint, and at the /users endpoint, I see user 1 has apps "app1" and "app3" associated with it.

When I enable this framework by adding the @EnableRelationalMongo annotation, the following exception is encountered upon hitting the /users endpoint. (It seems that the afterLoad() logic is querying for the related entities, but the _ids of the user's associated apps generated by the framework do not match the _ids of the apps actually persisted by spring.

```
java.lang.IllegalArgumentException: Collection must not be null or empty!
	at org.springframework.util.Assert.hasText(Assert.java:276) ~[spring-core-5.0.4.RELEASE.jar:5.0.4.RELEASE]
	at io.github.kaiso.relmongo.mongo.DatabaseLoader.getDocumentsById(DatabaseLoader.java:28) ~[relmongo-2.0.0-RC1.jar:na]
	at io.github.kaiso.relmongo.mongo.PersistentRelationResolver.resolveOnLoading(PersistentRelationResolver.java:54) ~[relmongo-2.0.0-RC1.jar:na]
	at io.github.kaiso.relmongo.events.listener.MongoEventListener.onAfterLoad(MongoEventListener.java:46) ~[relmongo-2.0.0-RC1.jar:na]
	at org.springframework.data.mongodb.core.mapping.event.AbstractMongoEventListener.onApplicationEvent(AbstractMongoEventListener.java:56) ~[spring-data-mongodb-2.0.5.RELEASE.jar:2.0.5.RELEASE]
	at org.springframework.data.mongodb.core.mapping.event.AbstractMongoEventListener.onApplicationEvent(AbstractMongoEventListener.java:31) ~[spring-data-mongodb-2.0.5.RELEASE.jar:2.0.5.RELEASE]
```

I'm using spring boot 2.0.2 managed dependencies and     

```
<dependency>
      <groupId>io.github.kaiso.relmongo</groupId>
      <artifactId>relmongo</artifactId>
      <version>2.0.0-RC1</version>
    </dependency>
```

**Comment by kaiso** [comment]
Hello Ok I'm trying to reproduce the issue

**Comment by kaiso** [comment]
@mancini0 
to fix your problem in the @Document annotation specify the collection name like this :

``@Document(collection = "collectionName")``

In the next release I will manage that it takes the default name from the Class name

**Comment by mancini0** [comment]
Cool, thanks for your quick responses. I'm looking forward to playing with this framework this weekend.

**Comment by kaiso** [comment]
Thank for using RelMongo, please feel free to submit issues.

I released a new version 2.0.0-RC2 which fix this issue, please close it if your tests are concluding.

**Comment by kaiso** [comment]
Fixed in 2.0.0-RC2
